grc/scripts/freedesktop/CMakeLists.txt

```diff
@@ -13,9 +13,14 @@ if(UNIX AND HAVE_XDG_UTILS AND ENABLE_POSTINSTALL)
     configure_file(
         ${CMAKE_CURRENT_SOURCE_DIR}/grc_setup_freedesktop.in
         ${CMAKE_CURRENT_BINARY_DIR}/grc_setup_freedesktop
+        @ONLY
+    )
+    file(
+        COPY ${CMAKE_CURRENT_SOURCE_DIR}/grc_setup_freedesktop.in
```

Comment:
Hm, that seems to be a bit wrong, in this order, the `file(COPY` should overwrite the `configure_file` result, right?
Wouldn't it make more sense to 

1. `configure_file(${CMAKE_CURRENT_SOURCE_DIR}/grc_setup_freedesktop.in` to `${CMAKE_CURRENT_BINARY_DIR}/grc_setup_freedesktop.tmp` (so, fill in the variables in the .in file, and put the result in build/…)
2. `file(COPY ${CMAKE_CURRENT_BINARY_DIR}/grc_setup_freedesktop.tmp ${CMAKE_CURRENT_BINARY_DIR}/grc_setup_freedesktop FILE_PERMISSIONS  …` to set the permissions?

(instead of doing the copy, it'd be more logical to do a `file(CHMOD…`, but that's a CMake 3.19 addition, and the fact that you don't have a very modern CMake was the original problem)

Reply:
There is always a better way. Let me test it and update you with the result.

Reply:
Re: Ubuntu: there's still a bug with us, if we need newer CMake than we declare that we need; see #7323 

Re: Better way: I think your approach here is wrong, not suboptimal :)